.github/workflows/cilium-integration-tests.yaml

```diff
@@ -0,0 +1,187 @@
+            })
+            
+            await github.rest.reactions.createForIssueComment({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              comment_id: ${{ github.event.comment.id }},
+              content: 'rocket'
+            })
+
+      - name: Checkout Cilium ${{ env.CILIUM_REPO_REF }}
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+        with:
+          repository: cilium/cilium # Be aware that this is the Cilium repository and not the one of the proxy itself!
+          ref: ${{ env.CILIUM_REPO_REF }}
+
+      - name: Install Cilium CLI ${{ env.CILIUM_CLI_TAG }}
+        run: |
+          cid=$(docker create quay.io/cilium/cilium-cli-ci:${{ env.CILIUM_CLI_TAG }} ls)
+          sudo docker cp $cid:/usr/local/bin/cilium /usr/local/bin
+          docker rm $cid
+          cilium version
+
+      - name: Create kind cluster
+        uses: helm/kind-action@d8ccf8fb623ce1bb360ae2f45f323d9d5c5e9f00 # v1.5.0
+        with:
+          version: ${{ env.KIND_VERSION }}
+          config: '.github/kind-config.yaml'
+          cluster_name: 'kind'
+
+      - name: Patch Cilium Agent Dockerfile
+        shell: bash
+        run: |
+          sed -i -E 's|(FROM )(quay\.io\/cilium\/cilium-envoy:)([0-9a-z]*)(@sha256:[0-9a-z]*)( as cilium-envoy)|\1${{ env.PROXY_IMAGE }}:${{ env.PROXY_TAG }}\5|' ./images/cilium/Dockerfile
+          cat ./images/cilium/Dockerfile
+
+      - name: Wait for Cilium Proxy image to be available
+        timeout-minutes: 30
+        shell: bash
+        run: until docker manifest inspect ${{ env.PROXY_IMAGE }}:${{ env.PROXY_TAG }} &> /dev/null; do sleep 15s; done
+
+      - name: Build Cilium Agent & Operator with patched Cilium Proxy Image
+        shell: bash
+        run: DOCKER_IMAGE_TAG=test make docker-cilium-image docker-operator-generic-image
+
+      - name: Load Cilium Images into kind
+        shell: bash
+        run: |
+          kind load docker-image \
+            --name kind \
+            quay.io/cilium/operator-generic:test \
+            quay.io/cilium/cilium:test
+
+      - name: Install Cilium
+        shell: bash
+        run: |
+          cilium install \
+            --chart-directory install/kubernetes/cilium \
+            --config monitor-aggregation=none \
+            --helm-set loadBalancer.l7.backend=envoy \
+            --helm-set tls.secretsBackend=k8s \
+            --agent-image=quay.io/cilium/cilium:test \
+            --operator-image=quay.io/cilium/operator-generic:test \
+            --helm-set image.pullPolicy=Never \
+            --helm-set operator.image.pullPolicy=Never \
+            --config debug=true \
+            --config debug-verbose=envoy
+
+          cilium hubble enable \
+            --chart-directory install/kubernetes/cilium
+
+          cilium hubble port-forward&
+
+      - name: Execute Cilium Connectivity Tests
+        shell: bash
+        run: cilium connectivity test
```

Comment:
IMO we could limit the tests with `--test=l7` option, that would limit the test to those with `l7` in their name, which would be all we need for Envoy integration. This would shave couple of minutes off the test time.

Reply:
Addressed with https://github.com/cilium/proxy/pull/165